go.mod

```diff
@@ -1,6 +1,6 @@
 module github.com/operator-framework/operator-controller
 
-go 1.23.4
+go 1.24.0
```

Comment:
/hold 

looks like updating this dep will require us to update to go 1.24, which we are not ready for yet